### PR TITLE
HLRC: Move ML protocol classes into client ml package

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**

### CODE REVIEW

**Review by droberts195 — APPROVED** [review]
LGTM

**Review by benwtrent — APPROVED** [review]
Green means GO! LTGM as long the build is green.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/RuleScope.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/RuleScope.java]: is this the only use of DeprecationHandler in the code? There is an instance of this handler in the RestHighLevelClient as well, so it might make more sense to use the one there?

**Review by hub-cap — APPROVED** [review]
one Q. Ill let you decide and not block the merge. I assume tho there will be more calls to `createParser` in the future, so we might want to use a common deprecation handler.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/RuleScope.java
Comment (dimitris-athanasiou) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/ml/job/config/RuleScope.java]: It is actually the only use. Though we could reuse the one in RestHighLevelClient, it is an odd dependency direction. I am tempted to duplicate for now. If we hit another we can refactor.

**Pull request merged**
